3.3/.sti/bin/assemble

```diff
@@ -1,23 +1,66 @@
 #!/bin/bash
 
-# For SCL enablement
-source .bashrc
+# This script uses these environment variables:
+#
+# DISABLE_COLLECTSTATIC: if not empty, inhibits execution of 'manage.py collectstatic'.
 
-set -e
+function assemble() {
+  # For SCL enablement
+  source .bashrc
```

Comment:
@mfojtik 
> you want to have this globally enabled

For both scripts, the way I thought about it, is that the `assemble` function is the entry point of this script. I wanted to read from top to bottom and be able to understand and explain what is going to happen all the time without jumping in the file -- trying to get us a little closer to literate programming? Instead of having some commands, then function declarations and then more commands, and have to read everything and jump top-bottom-top to understand what is going on.

The call in or outside the function has the same "global" effect, as it affects the current shell.

But I understand perhaps I should accommodate to whatever style and tradition is already in place...

> sourcing files in functions is bad practice because it makes it hard to test

Agree, for all functions that are not `main`... maybe I should call it `main()` and not `assemble()`?! :smile: 
I'd avoid `source`'ing inside a function just like avoid changing shell options with `set`.

> also if you call this function twice, you will enable SCL twice...

`source scl_source python33` is idempotent, calling it multiple times makes no harm (won't duplicate entries in PATH, etc).